### DISCUSSION TITLE
Consider switching MDNS container for hostname container

### DISCUSSION THREAD

**Discussion by maggie44** [maggie0002/balena-device-ui#61] [Ideas]
Considering the size and complexity of the MDNS container, and the fact that it isn't going to be needed by a lot of users, is it worth switching it out for the hostname container (https://github.com/balenablocks/hostname) as the default, and then let users add the MDNS if they require it? The hostname container will change the hostname to a specified name, which will then make the device accessible via `the-specified-name.local`. The only reason for then needing an additional MDNS is if that hostname is going to be changing, which is quite a niche use case.

Hostname container is around 10mb and MDNS is 200mb+. It is also uses less resources and exits when finished.

@nucleardreamer

**Comment by 0xVANISHED** [comment]
So there are a couple differences using the MDNS container vs changing just the hostname

- Pro: Like you said, very small image size (and honestly we don't even have to use it, if we just call the hostname endpoint with the API)
- Con: Changing the hostname will persist across deployments, so it might be slightly unexpected experience for someone to run our docker-compose, push different software, and be stuck on the previous hostname (they will have to then use the hostname block or supervisor API to switch to something else).
- Con: Changing the hostname to get the new MDNS advertisement also requires a reboot of the avahi service to pickup the new change, which is usually done by just rebooting the device (or restarting the avahi service via systemctl on the hostOS).

The `mdns-advertise` container takes care of these cons by setting avahi to advertise programmatically and in a non-destructive way, but the container size is much bigger.

I think we can go either way for sure, as long as we know that there might be a potentially more bumpy experience using either the `hostname` block or just hitting the endpoint on the supervisor

**Reply by maggie44** [reply]
You're right about calling the endpoints rather than using the block, we had the endpoints already. It is added with a environment variable so if someone didn't want to change the hostname they can just remove the env variable from the compose. Makes it completely optional.
